bfdd/bfd.c

```diff
@@ -400,12 +400,21 @@ struct bfd_session *ptm_bfd_sess_find(struct bfd_pkt *cp,
 	if (cp->discrs.remote_discr)
 		return bfd_find_disc(peer, ntohl(cp->discrs.remote_discr));
 
-	/* Search for session without using discriminator. */
-	ifp = if_lookup_by_index(ifindex, vrfid);
-	if (vrfid != VRF_DEFAULT)
-		vrf = vrf_lookup_by_id(vrfid);
-	else
-		vrf = NULL;
+	/*
+	 * Search for session without using discriminator.
+	 *
+	 * XXX: we can't trust `vrfid` because the VRF handling is not
+	 * properly implemented. Meanwhile we should use the interface
+	 * VRF to find out which one it belongs.
+	 */
+	ifp = if_lookup_by_index_all_vrf(ifindex);
```

Comment:
The function if_lookup_by_index_all_vrf() makes an assumption that one ifp can belong to only one VRF. Although I think this assumption is valid. (I can't think of an layer 3 interface which can belong to multiple VRF). You can cross check.

Reply:
Interface name might be available in multiple VRFs (e.g. `lo`), however I think this might not be the case for indexes.